### PR TITLE
fix: docker compose

### PR DESCRIPTION
**Pull request by DerekRoberts**
# Description

Run docker compose without needing to specify envars.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manually.  Further help required.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!
Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-backend-70-backend.apps.silver.devops.gov.bc.ca/) available
Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-backend/actions/workflows/merge-main.yml)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -18,24 +17,23 @@ services:
...
   backend:
     container_name: backend
-    image: ghcr.io/bcgov/nr-spar-backend-backend:test
     depends_on:
       database:
         condition: service_healthy
     environment:
       NR_SPAR_BACKEND_VERSION: snapshot
-      KEYCLOAK_REALM_URL: ${KEYCLOAK_REALM_URL}
+      KEYCLOAK_REALM_URL: https://dev.loginproxy.gov.bc.ca/auth/realms/standard
```
Comment (RMCampos) [inline docker-compose.yml:32]: We stopped using `dev` keycloak server, only `test` now. Can you update here?

**Review by RMCampos — DISMISSED** [review]
Well done! Once keycloak server url gets updated we should be good to go.

**Review by annibalsilva — DISMISSED** [review]
I thought about this change over last night and I'd like to keep using environment variables for the database properties.

I (and I believe it's the case with @RMCampos as well) am not dealing with any other local database at the moment, so the change would be fine for now; but a time might come when we need to a parallel version of this application's database, for instance, and the hard coded "default" would be a hindrance.

Because of that, I suggest keeping the environment variables for the database and attributing default values to them so we don't _need_ to use environment variables, but we can if we want or need to.

**Comment by DerekRoberts** [comment]
> I thought about this change over last night and I'd like to keep using environment variables for the database properties.
> 
> I (and I believe it's the case with @RMCampos as well) am not dealing with any other local database at the moment, so the change would be fine for now; but a time might come when we need to a parallel version of this application's database, for instance, and the hard coded "default" would be a hindrance.
> 
> Because of that, I suggest keeping the environment variables for the database and attributing default values to them so we don't _need_ to use environment variables, but we can if we want or need to.

Do not use envars for Docker Compose.  That's bad practice and increases barriers for entry.

**Review by annibalsilva — APPROVED** [review]

**Pull request merged**